### PR TITLE
Initial Commit of ACS data source pulling data for state level for race and sex of health insurance.

### PR DESCRIPTION
**Pull request by Sethvg**
This pull request is a milestone, and while not end to end, I want to get feedback now so that I dont overwhelm with one large pull request.

### CODE REVIEW

File: python/datasources/acs_health_insurance.py [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+#   ...
+# }
+# to create the ACS variable params.
+# ?for=state&get=C27001A_002E,C27001A_003E...
+
+
+def format_params(prefixes, suffixes, isCounty=False):
+    groups = []
+    for prefix in prefixes:
+        for suffix in suffixes:
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:199]: You can make all of this into a one-liner using list comprehension if you want:

groups = [p + s for p in prefixes for s in suffixes]

File: python/datasources/acs_health_insurance.py [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+# {
+#   "_002E":meta(None,0, 19),
+#   "_003E":meta(None,0, 19,HealthInsurancePopulation.WITH)
+#   ...
+# }
+# to create the ACS variable params.
+# ?for=state&get=C27001A_002E,C27001A_003E...
+
+
+def format_params(prefixes, suffixes, isCounty=False):
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:196]: nit: can you change isCounty to is_county to match python style guide / your naming elsewhere in this file for method params

File: python/datasources/acs_health_insurance.py [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+    groups = []
+    for prefix in prefixes:
+        for suffix in suffixes:
+            groups.append(prefix+suffix)
+    vars = ','.join(groups)
+
+    return {'for': 'county' if isCounty else 'state', "get": vars}
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:203]: another nit: it would be nice to either use double quotes everywhere or single quotes everywhere in the file

File: python/datasources/acs_health_insurance.py [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+    # Initialize variables in class instance, also merge all metadata so that lookup of the
+    # prefix, suffix combos can return the entire metadata
+    def __init__(self, base):
+        self.baseUrl = base
+        self.data = {}
+        self.metadata = {}
+        self.buildMetadataList()
+
+    # Gets standardized filename
+    def getFilename(self, sex, race, isCounty):
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:225]: nit: isCounty -> is_county

also the python style guide (at least within google) recommends underscored names for methods rather than camel case, e.g. get_filename instead of getFilename.. would you mind updating your methods to match that?

File: python/datasources/acs_health_insurance.py [hunk 5]
```diff
@@ -0,0 +1,657 @@
...
+        if(race is not None):
+            return "HEALTH_INSURANCE_BY_RACE_{0}_{1}.json".format("STATE" if isCounty else "COUNTY", race)
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:227]: It's odd that you do "STATE" if isCounty else "County" here and right below.. should is_county be renamed to is_state ?

File: python/datasources/acs_health_insurance.py [hunk 6]
```diff
@@ -0,0 +1,657 @@
...
+    # Method to output <Filename.csv>.  Used for debugging purposes.
+    def write_local_files_debug(self):
+        self.getStateFipsMap()
+        self.getCountyFipsMap()
+        self.getHealthInsuranceDataBySex()
+        self.getHealthInsuranceDataByRace()
+        self.splitDataFrames()
+
+        with open('total_health_insurance.json', 'w') as f:
+            print(json.dumps(
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:240]: nit: i think it's more idiomatic to do f.write(json.dumps(...)) for writing to a file. Also, please call f.close() once you are finished with it

File: python/datasources/acs_health_insurance.py [hunk 7]
```diff
@@ -0,0 +1,657 @@
...
+    # Uploads the ACS data to GCS by providing
+    # the ACS Base URL
+    # Acs Query Params
+    # Standardized Filename
+    #
+    # An example file created in GCS:
+    # HEALTH_INSURANCE_BY_RACE_COUNTY_WHITE_ALONE.json
+
+    def upload_to_gcs(self, bucket):
+        maleStateParams = format_params(
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:255]: I believe the python style guide (at least within google) says to use underscored names for all local variables rather than camel case, e.g. male_state_params instead of maleStateParams. Would you mind updating all of your local variables to follow this style?

File: python/datasources/acs_health_insurance.py [hunk 8]
```diff
@@ -0,0 +1,657 @@
...
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, maleStateParams, bucket,
+            self.getFilename(Sex.MALE, None, False))
+
+        femaleStateParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:263]: This operation is a bit hard to parse, could you put the "file_diff and" part at the beginning to make it clearer, e.g. file_diff = file_diff and url_file_to_gcs...

ditto for the same operation right below

File: python/datasources/acs_health_insurance.py [hunk 9]
```diff
@@ -0,0 +1,657 @@
...
+            self.baseUrl, femaleStateParams, bucket,
+            self.getFilename(Sex.FEMALE, None, False)) and fileDiff
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:265]: When calling getFilename, it's best practice for readability to explicitly name the arguments being passed which aren't variables, in this case it would self.getFilename(Sex.Filename, race=None, is_county=False). Could you make this change here and in other places where you have floating constants such as None/False/True being passed to methods?

File: python/datasources/acs_health_insurance.py [hunk 10]
```diff
@@ -0,0 +1,657 @@
...
+    # Acs Query Params
+    # Standardized Filename
+    #
+    # An example file created in GCS:
+    # HEALTH_INSURANCE_BY_RACE_COUNTY_WHITE_ALONE.json
+
+    def upload_to_gcs(self, bucket):
+        maleStateParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:257]: It would be a helpful to add an comment explaining what fileDiff intends to accomplish

File: python/datasources/acs_health_insurance.py [hunk 11]
```diff
@@ -0,0 +1,657 @@
...
+            self.baseUrl, maleCountyParams, bucket,
+            self.getFilename(Sex.MALE, None, True)) and fileDiff
+
+        femaleCountyParams = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES, True)
+        fileDiff = url_file_to_gcs.url_file_to_gcs(
+            self.baseUrl, femaleCountyParams, bucket,
+            self.getFilename(Sex.FEMALE, None, True)) and fileDiff
+
+        for prefix in HEALTH_INSURANCE_BY_RACE_GROUP_PREFIXES:
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:279]: It would be helpful to add a comment above this block explaining what it does

File: python/datasources/acs_health_insurance.py [hunk 12]
```diff
@@ -0,0 +1,657 @@
...
+        self.getCountyFipsMap()
+
+        # Pull data from GCS and aggregate in memory
+        self.getHealthInsuranceDataBySex(useGcs=True, gcs_bucket=gcs_bucket)
+        self.getHealthInsuranceDataByRace(useGcs=True, gcs_bucket=gcs_bucket)
+
+        # Split internal memory into data frames for sex/race by state/county
+        self.splitDataFrames()
+
+        # Create BQ collumns and write dataframes to BQ
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:307]: nit, typo: s/collumns/columns

File: python/datasources/acs_health_insurance.py [hunk 13]
```diff
@@ -0,0 +1,657 @@
...
+            countyFip = data[1]
+            age = data[2]
+            sex = data[3]
+            race = data[4]
+            population = self.data[data]
+            whi = population[HealthInsurancePopulation.WITH]
+            wohi = population[HealthInsurancePopulation.WITHOUT]
+            total = population[HealthInsurancePopulation.TOTAL]
+
+            #
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:561]: dangling comment?

File: python/datasources/acs_health_insurance.py [hunk 14]
```diff
@@ -0,0 +1,657 @@
...
+        countySexData = []
+
+        # Extract keys from self.data Tuple
+        # (StateFip, CountyFip, Age, Sex, Race): {PopulationObj}
+        for data in self.data:
+            stateFip = data[0]
+            countyFip = data[1]
+            age = data[2]
+            sex = data[3]
+            race = data[4]
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:555]: A simpler way to do this expansion (assuming data only contains 5 elements) is:

state_fip, county_fip, age, sex, race = data

**Review by vanshkumar — COMMENTED** [review]
Sorry for all of the comments! Most of them are formatting related and shouldn't be too hard to resolve. Just to call a couple out at top-level:
- If you could make all of your lines shorter than 80 chars, that would be appreciated
- If you could make your variable and method naming conform to the [Google Python style guide](https://g3doc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#316-naming), that would be amazing

File: .gitignore [hunk 15]
```diff
@@ -22,3 +22,17 @@ node_modules/
 /frontend/.env.test.local
 /frontend/.env.production.local
 /frontend/npm-debug.log*
+
+health_insurance_by_race_state.json
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply .gitignore:26]: This is the git ignore file.  Its just so if it gets created it doesnt get added to the repo.  I changed it to .*json and .*csv for all test files in the root dir.

File: python/datasources/acs_health_insurance.py [hunk 16]
```diff
@@ -0,0 +1,657 @@
...
+                                            STATE_NAME_COL, COUNTY_NAME_COL,
+                                            AGE_COL, SEX_COL,
+                                            RACE_COL, WITH_HEALTH_INSURANCE_COL,
+                                            WITHOUT_HEALTH_INSURANCE_COL,
+                                            TOTAL_HEALTH_INSURANCE_COL)
+# TODO pass this in from message data.
+BASE_ACS_URL = "https://api.census.gov/data/2019/acs/acs5"
+
+
+class HealthInsurancePopulation:
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:18]: Intellisense mostly.  It acts as an enum and you can goto defintion of the variable rather than if you did HEALTH_INSURANCE_POPULATION["WITH" you could only go to the dict defintion.  This doesnt scale well.  Also I think the classes look much cleaner.

File: python/datasources/acs_health_insurance.py (hunk 10 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:257]: Ok I added a snippet

File: python/datasources/acs_health_insurance.py (hunk 11 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:279]: Done

File: python/datasources/acs_health_insurance.py (hunk 12 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:307]: fixed

File: python/datasources/acs_health_insurance.py (hunk 14 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:555]: Thats cool!

File: python/datasources/acs_health_insurance.py [hunk 17]
```diff
@@ -0,0 +1,657 @@
...
+    WITH = "With"
+    WITHOUT = "Without"
+    TOTAL = "Total"
+
+
+class Race:
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:24]: Okay I changed this one to be Race as a dict.

File: python/ingestion/url_file_to_gcs.py [hunk 18]
```diff
@@ -97,6 +97,7 @@ def download_first_url_to_gcs(url_list, gcs_bucket, dest_filename,
     if files_are_diff:
         # Upload the contents to the bucket
         bucket.blob(dest_filename).upload_from_filename(new_file_local_path)
+        print(f'Uploading to Gcs_Bucket: {gcs_bucket}, FileName: {dest_filename}')
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/ingestion/url_file_to_gcs.py:100]: removed

File: .gitignore (hunk 15 above)
[the remark this answers is not in the document]
Comment (vanshkumar) [reply .gitignore]: SG, the * method seems better than explicitly calling out files

File: python/datasources/acs_health_insurance.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (vanshkumar) [reply python/datasources/acs_health_insurance.py:18]: Fair enough; in that case, do you want to explicitly subclass Enum for these so it's clearer?

File: python/datasources/acs_health_insurance.py [hunk 19]
```diff
@@ -0,0 +1,612 @@
...
+    def write_to_bq(self, dataset, gcs_bucket):
+        # Get an ACS mapping of Fip Codes to State Names and county codes to county names
+        self.get_state_fips_mapping()
+        self.get_county_fips_mapping()
+
+        # Pull data from GCS and aggregate in memory
+        self.get_health_insurance_data_by_sex(
+            use_gcs=True, gcs_bucket=gcs_bucket)
+        self.get_health_insurance_data_by_race(
+            use_gcs=True, gcs_bucket=gcs_bucket)
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:279]: What's the reason that you don't also aggregate by age? Is that not possible with this data? I was imagining that 1-D aggregations by race, age, and sex would be doable

File: python/datasources/acs_health_insurance.py [hunk 20]
```diff
@@ -0,0 +1,612 @@
...
+    # EG:
+    # Prefixes:
+    # {"C27001A":{MetadataKey.RACE: RACE["WHITE_ALONE"]}},
+    # {"C27001B":{MetadataKey.RACE: RACE["BLACK_OR_AFRICAN_AMERICAN_ALONE"]}}
+    # Suffixes:
+    #    {"_002E":meta(None,0, 19),
+    #     "_003E":meta(None,0, 19,HealthInsurancePopulation.WITH) ``}
+    # Combine Into Four:
+    # {"C27001A_002E": {Race: White, Age: 0-19: Sex: None, Population: Total}}
+    # {"C27001A_OO3E": {Race: White, Age: 0-19: Sex: None, PopulW[MetadataKey.RACE]
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:309]: Is the "PopulW" part at the end of this comment correct?

File: python/datasources/acs_health_insurance.py [hunk 21]
```diff
@@ -0,0 +1,612 @@
...
+                    HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES[female_suffix])
+                self.metadata[key] = meta
+
+    # Pull the State_Fips map Code->Name from ACS
+
+    def get_state_fips_mapping(self):
+        params = {'for': 'state', "get": "NAME"}
+        resp = requests.get(self.base_url, params=params)
+        json_formatted_response = resp.json()
+        state_fips = {}
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:340]: I think you can get rid of the state_fips local variable and directly modify self.state_fips

ditto for county_fips in the method right below this one

File: python/datasources/acs_health_insurance.py [hunk 22]
```diff
@@ -0,0 +1,612 @@
+import json
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:1]: In general, the logic was a bit hard to follow for me.. could you add a multi-line comment at the top of the file describing at a high-level what this code is supposed to do & what options are available (eg local vs on gcp)

File: python/datasources/acs_health_insurance.py [hunk 23]
```diff
@@ -0,0 +1,612 @@
...
+
+    # Get Health insurance By Sex from either API or GCS and aggregate it in memory
+    def get_health_insurance_data_by_sex(self, use_gcs=False, gcs_bucket=None):
+        if(use_gcs):  # LOAD JSON BLOBS FROM GCS
+            male_state_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.MALE, None, False))
+            female_state_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.FEMALE, None, False))
+            male_county_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.MALE, None, True))
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:399]: It would be great if you could add the name for arguments which are "hanging" e.g. change self.get_filename(Sex.Male, None, False) to self.get_filename(Sex.Male, race=None, is_county=False). It would be amazing if you could do this throughout the file wherever you have these hanging arguments, it helps quite a bit with readability / being able to follow the flow of the logic imo

File: python/datasources/acs_health_insurance.py [hunk 24]
```diff
@@ -0,0 +1,612 @@
...
+            female_county_data = self.get_acs_data_from_variables(
+                female_county_request_params)
+
+        # Aggregate and accumulate data in memory
+        self.accumulate_acs_data(male_state_data)
+        self.accumulate_acs_data(female_state_data)
+        self.accumulate_acs_data(male_county_data)
+        self.accumulate_acs_data(female_county_data)
+
+    '''
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:427]: Could you move this large comment (and others in the file) inside the method they apply to, and describe the inputs/outputs? E.g. see [here](https://g3doc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#function-docs)

File: python/datasources/acs_health_insurance.py [hunk 25]
```diff
@@ -0,0 +1,612 @@
...
+    Takes data in the form of
+    [
+        [C27001A_002E, C27001A_003E, C27001B_002E, C27001B_002E][state][?county],
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:430]: Is the "?" in "?county" supposed to be there?

File: python/datasources/acs_health_insurance.py [hunk 26]
```diff
@@ -0,0 +1,612 @@
...
+        ...
+    } Note: This can be debugged via the total_health_insurance.json output file via local_debug
+    '''
+
+    def accumulate_acs_data(self, data):
+        key_row = data[0]
+        for row in data[1::]:
+            data = {}
+            for key in key_row:
+                if(key != 'state' and key != 'county'):
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:459]: You can remove the parens in the if statement here. Also, should this quit with some kind of error? IIUC we should only have state or county data here, so not logging & exiting seems reasonable if the key isn't one of those two

File: python/datasources/acs_health_insurance.py [hunk 27]
```diff
@@ -0,0 +1,612 @@
...
+            "Without": "13947"
+        },
+        ...
+    } Note: This can be debugged via the total_health_insurance.json output file via local_debug
+    '''
+
+    def accumulate_acs_data(self, data):
+        key_row = data[0]
+        for row in data[1::]:
+            data = {}
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:457]: It would be clearer to use a different name for the local variable "data", as it's already the name of an argument passed into this method

File: python/datasources/acs_health_insurance.py [hunk 28]
```diff
@@ -0,0 +1,612 @@
...
+            for key in key_row:
+                if(key != 'state' and key != 'county'):
+                    data[key] = {}
+            state_fip = None
+            county_fip = None
+            col_index = 0
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:463]: Can you get rid of col_index and instead just loop over the number of elements in row? E.g.:

for col_index in range(len(row)):
  ...

File: python/datasources/acs_health_insurance.py [hunk 29]
```diff
@@ -0,0 +1,612 @@
...
+            for col in row:
+                key = key_row[col_index]
+                if(key == 'state'):
+                    state_fip = col  # Extract the static key_row state
+                elif key == 'county':
+                    # Extract the static key_row (county) *if exists
+                    county_fip = col
+                else:
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:471]: When would this else statement trigger?

File: python/datasources/acs_health_insurance.py [hunk 30]
```diff
@@ -0,0 +1,612 @@
...
+                    metadata.get(MetadataKey.RACE))
+                row[metadata[MetadataKey.POPULATION]] = value
+
+    # Helper method from grabbing a tuple in self.data.  If the
+    # tuple hasnt been created then it initializes an empty tuple.
+    # This is needed as each data variable will only
+    # update one of the population values at a time.
+
+    def upsert_row(self, state_fip, county_fip, age, sex, race):
+        if self.data.get((state_fip, county_fip, age, sex, race)) is None:
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:490]: A simpler way of doing this key presence check is:

if (state_fip, county_fip, age, sex, race) is not in self.data:
  ...

gotta love python :)

File: python/datasources/acs_health_insurance.py [hunk 31]
```diff
@@ -0,0 +1,612 @@
...
+                    county_fip = col
+                else:
+                    data[key] = {'value': col, 'meta': self.metadata[key]}
+
+                col_index += 1
+
+            for var in data:
+                metadata = data[var]['meta']
+                value = data[var]['value']
+                row = self.upsert_row(state_fip, county_fip, metadata.get(
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:479]: Rather than doing "metdata.get(...)" you should just be able to index directly, eg "metadata[...]"

File: python/datasources/acs_health_insurance.py [hunk 32]
```diff
@@ -0,0 +1,612 @@
...
+        for ingester in self._create_ingesters():
+            next_file_diff = ingester.upload_to_gcs(gcs_bucket)
+            file_diff = file_diff or next_file_diff
+        return file_diff
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+        for ingester in self._create_ingesters():
+            ingester.write_to_bq(dataset, gcs_bucket)
+
+    def _create_ingesters(self):
```
Comment (vanshkumar) [inline python/datasources/acs_health_insurance.py:603]: Is this method necessary as it only returns a single element list?

File: python/datasources/acs_health_insurance.py [hunk 33]
```diff
@@ -0,0 +1,612 @@
...
+
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, male_state_params, bucket,
+            self.get_filename(Sex.MALE, None, False))
+
+        female_state_params = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES)
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, female_state_params, bucket,
+            self.get_filename(Sex.FEMALE, None, False)) and file_diff
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:237]: Done

File: python/datasources/acs_health_insurance.py [hunk 34]
```diff
@@ -0,0 +1,612 @@
...
+    # prefix, suffix combos can return the entire metadata
+    def __init__(self, base):
+        self.base_url = base
+        self.data = {}
+        self.metadata = {}
+        self.build_metadata_list()
+
+    # Gets standardized filename
+    def get_filename(self, sex, race, is_county):
+        if(race is not None):
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:194]: done

File: python/datasources/acs_health_insurance.py [hunk 35]
```diff
@@ -0,0 +1,612 @@
...
+    # Initialize variables in class instance, also merge all metadata so that lookup of the
+    # prefix, suffix combos can return the entire metadata
+    def __init__(self, base):
+        self.base_url = base
+        self.data = {}
+        self.metadata = {}
+        self.build_metadata_list()
+
+    # Gets standardized filename
+    def get_filename(self, sex, race, is_county):
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:193]: added

File: python/datasources/acs_health_insurance.py (hunk 19 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:279]: I could aggregate by age, but the issue requirements at the end we would end up with 4 files, and this is how it was spelled out in the issue.  Let me know if this is needed.

File: python/datasources/acs_health_insurance.py (hunk 28 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:463]: Okay

File: python/datasources/acs_health_insurance.py (hunk 29 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:471]: This else statement is the most triggered.  The data is the format [var_1, var_2, ..., var_n, state, county].  This else is for every var

File: python/datasources/acs_health_insurance.py (hunk 31 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:479]: This is not the case, as race and sex are not gaurenteed to be set and will trigger a Key Error

File: python/datasources/acs_health_insurance.py (hunk 32 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:603]: Im just following the flow from ACS_population, and could be split out in the future

File: python/datasources/acs_health_insurance.py [hunk 36]
```diff
@@ -0,0 +1,612 @@
...
+    # Uploads the ACS data to GCS by providing
+    # the ACS Base URL
+    # Acs Query Params
+    # Standardized Filename
+    #
+    # An example file created in GCS:
+    # HEALTH_INSURANCE_BY_RACE_COUNTY_WHITE_ALONE.json
+    #
+    # Returns:
+    # FileDiff = If the data has changed by diffing the old run vs the new run.
```
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:223]: THis is following the pattern of the ACS_Population flow.  I will schedule a meeting but should proceed with it in now and if need to circle back should be changes similar to ACS pop.

File: python/datasources/acs_health_insurance.py (hunk 30 above)
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:490]: This throws a syntax error :/

File: python/datasources/acs_health_insurance.py (hunk 19 above)
Comment (vanshkumar) [reply python/datasources/acs_health_insurance.py:281]: Just read the issue and it looks like it's because the data is organized as such (age/race and age/sex). I think the data we eventually want to show in the FE is broken down each by race, age, and sex (i.e. 1-D breakdowns), but having these tables is fine as that is computable from these.

File: python/datasources/acs_health_insurance.py (hunk 29 above)
Comment (vanshkumar) [reply python/datasources/acs_health_insurance.py:473]: I see, thanks for the explanation

File: python/datasources/acs_health_insurance.py (hunk 31 above)
Comment (vanshkumar) [reply python/datasources/acs_health_insurance.py:479]: I see

File: python/datasources/acs_health_insurance.py (hunk 30 above)
Comment (vanshkumar) [reply python/datasources/acs_health_insurance.py:490]: Oops, I accidentally included "is" in there, it should be:

if (state_fip, county_fip, age, sex, race) not in self.data:
...

Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:491]: All good, updated

File: python/datasources/acs_health_insurance.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (Sethvg) [reply python/datasources/acs_health_insurance.py:19]: Subclassing Enum in python is wierd

**Review by vanshkumar — APPROVED** [review]
LGTM modulo remaining open comments
